Active-Directory/Backup-GroupPolicy.ps1

```diff
@@ -63,15 +63,26 @@ param (
 
     [Parameter()]
     [string]
-    $Domain = (Get-WmiObject Win32_ComputerSystem).Domain,
-
-    [Parameter()]
-    [string]
+    $Domain = $env:USERDNSDOMAIN,
+
+    # Specify aliases for the Server parameter and support tab completion for domain controller names.
+    [Parameter(]
+    [Alias("DomainController","DC")]
+    [ArgumentCompleter( {
+        param ( $commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameters )
+        $possibleValues = @{ Server = (Get-ADDomainController -Filter *).Hostname }
+        if ($fakeBoundParameters.ContainsKey('Type')) { $possibleValues[$fakeBoundParameters.Type] | Where-Object { $_ -like "$wordToComplete*" } }
+        else { $possibleValues.Values | ForEach-Object {$_} }
+    } )]
```

Comment:
This change will instantly get the current domain name from system environment variables instead of processing a WMI query.